test/cpp/api/transformer.cpp

```diff
@@ -460,3 +430,492 @@ TEST_F(TransformerTest, PrettyPrintTransformerDecoderLayer) {
+  TransformerDecoder multi_layer_model(
+    TransformerDecoderOptions(decoder_layer, 2));
+  if (is_cuda) {
+    model->to(torch::kCUDA);
+  }
+
+  decoder_input = torch::tensor({{{20, 30, 40, 50}}}, tensor_options);
+  memory_input = torch::tensor({{{60, 70, 80, 90}}}, tensor_options);
+  result = multi_layer_model(decoder_input, memory_input).detach();
+  ref_output = torch::tensor(
+    {{{2.31316, 0.0950293, -0.671995, 0.102802}}},
+    tensor_options);
+  ASSERT_EQ(result.sizes().size(),ref_output.sizes().size());
+  ASSERT_TRUE(torch::allclose(result, ref_output, 1e-7, 1e-5,
+                              /*equal_nan=*/true));
+
+  // multiple layers no norm
+  TransformerDecoder multi_layer_model2(
+    TransformerDecoderOptions(decoder_layer, 6));
+  if (is_cuda) {
+    model->to(torch::kCUDA);
+  }
+   // deterministic input
+  decoder_input = torch::tensor({{{0.4517, 0.6793, 0.5313, 0.0034},
+                                  {0.2678, 0.3677, 0.4459, 0.7166}},
+                                 {{0.8100, 0.3716, 0.4096, 0.1976},
+                                  {0.6958, 0.8844, 0.6081, 0.8315}},
+                                 {{0.0494, 0.9343, 0.5955, 0.3830},
+                                  {0.5404, 0.3464, 0.9378, 0.6200}}},
+                                  tensor_options);
+  memory_input = torch::tensor({{{0.7462, 0.6653, 0.5679, 0.4891},
+                                 {0.5387, 0.1655, 0.3565, 0.0471}},
+                                {{0.8335, 0.2799, 0.5031, 0.2947},
+                                 {0.1402, 0.0318, 0.7636, 0.1346}},
+                                {{0.6333, 0.9344, 0.1376, 0.9938},
+                                 {0.8924, 0.2872, 0.6692, 0.2944}},
+                                {{0.9897, 0.6915, 0.3154, 0.1733},
+                                 {0.8645, 0.3513, 0.3064, 0.0767}},
+                                {{0.8117, 0.2366, 0.4838, 0.7881},
+                                 {0.3718, 0.4945, 0.9511, 0.0864}}},
+                                 tensor_options);
+  result = multi_layer_model2(decoder_input, memory_input).detach();
+  ref_output = torch::tensor({{{2.42794, 0.026164, -0.60263, -0.0747591},
+                               {2.43113, 0.0279516, -0.600376, -0.0736896}},
+                              {{2.42794, 0.026164, -0.60263, -0.0747591},
+                               {2.43113, 0.0279516, -0.600376, -0.0736896}},
+                              {{2.42794, 0.026164, -0.60263, -0.0747591},
+                               {2.43113, 0.0279516, -0.600376, -0.0736896}}},
+                               tensor_options);
+  ASSERT_EQ(result.sizes().size(),ref_output.sizes().size());
+  ASSERT_TRUE(torch::allclose(result, ref_output, 1e-7, 1e-5,
+                              /*equal_nan=*/true));
+
+
+  // multiple layers with norm
+  LayerNorm norm(LayerNormOptions({decoder_layer.get()->options.d_model()}));
+  TransformerDecoder multi_layer_norm_model(
```

Comment:
Same here, this inconsistency caused illegal memory access in the layer ASSERT_TRUE comparison.
I recommend always use `model = TransformerDecoder(...)`